front/src/applications/operationalStudies/views/ManageTimetableItem.tsx

```diff
@@ -228,14 +229,42 @@ const ManageTimetableItem = () => {
     []
   );
 
-  const showCategoryWarning =
-    rollingStock &&
-    currentCategory &&
-    isMainCategory(currentCategory) &&
-    currentCategory.main_category !== rollingStock.primary_category &&
-    !rollingStock.other_categories.includes(currentCategory.main_category);
-  const categoryWarning = showCategoryWarning ? t('categoryMismatch') : undefined;
+  const { data } = osrdEditoastApi.endpoints.getSubCategory.useQuery({
+    page: 1,
+    pageSize: 100,
+  });
+
+  // todo: remove mocked data when backend is ready
+  const subCategoriesFromApiOrMocked =
+    data && data.results && data.results.length > 0 ? data.results : subCategoriesMocked;
+
+  const currentSubCategory = useMemo(() => {
+    if (currentCategory && !isMainCategory(currentCategory)) {
+      return subCategoriesFromApiOrMocked.find(
+        (option) => option.code === currentCategory.sub_category_code
+      );
+    }
+    return undefined;
+  }, [currentCategory, subCategoriesFromApiOrMocked]);
```

Comment:
Would be cleaner with an early return
```suggestion
  const currentSubCategory = useMemo(() => {
    if (!currentCategory || isMainCategory(currentCategory)) return undefined;

    return subCategoriesFromApiOrMocked.find(
      (option) => option.code === currentCategory.sub_category_code
    );
  }, [currentCategory, subCategoriesFromApiOrMocked]);
```

Also `isMainCategory` should also handle `null` since a main category can also be null.
```
export function isMainCategory(category: TrainCategory | null) {
  return category === null || 'main_category' in category;
}
```
You can then avoid some checks everywhere you call `isMainCategory` like the one above.

Reply:
Done